base/array.jl

```diff
@@ -237,16 +237,29 @@ function _collect(cont, itr, ::HasEltype, isz::SizeUnknown)
     return a
 end
 
+if isdefined(:Base)
+@eval function _default_eltype(itrt::ANY)
+    $(Expr(:meta, :pure))
+    try
+        rt = return_types(first, (itrt,))
```

Comment:
I hear you. But I think, at the very least, `return_type` is a better, more general interface to it.